test/test_torch.py

```diff
@@ -5307,6 +5307,52 @@ def test_tensor_grad_warnings(self):
             foo = dummy.grad
             self.assertEqual(len(w), 1)
 
+    def test_normal_shape(self):
+        for device in torch.testing.get_all_device_types():
+            tensor1 = torch.rand(1, device=device)
+            tensor4 = torch.rand(4, device=device)
+            tensor120 = torch.rand(120, device=device)
+            tensor2145 = torch.rand(2, 1, 4, 5, device=device)
+            tensor2345 = torch.rand(2, 3, 4, 5, device=device)
+            tensor2345_non_contiguous = torch.rand(2, 4, 3, 5, device=device).permute(0, 2, 1, 3)
+            tensor2345_channels_last = tensor2345.contiguous(memory_format=torch.channels_last)
+            output2345 = torch.zeros(2, 3, 4, 5, device=device)
+            output345 = torch.zeros(3, 4, 5, device=device)
+
+            # inputs have same size
+            self.assertEqual(torch.normal(tensor2345, tensor2345).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2345_non_contiguous, tensor2345).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2345, tensor2345_channels_last).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2345_non_contiguous, tensor2345_channels_last).size(), (2, 3, 4, 5))
+
+            # scalar case
+            self.assertEqual(torch.normal(tensor2345, 2).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(2, tensor2345).size(), (2, 3, 4, 5))
+
+            # inputs are expandable tensors
+            self.assertEqual(torch.normal(tensor2345, tensor1).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2145, tensor2345).size(), (2, 3, 4, 5))
+
+            # inputs are non-expandable tensors, but they have same number of elements
+            self.assertEqual(torch.normal(tensor120, tensor2345).size(), (120,))
+            self.assertEqual(torch.normal(tensor2345, tensor120).size(), (2, 3, 4, 5))
```

Comment:
If possible, we should also check that in this case (inputs have same number of elements) the deprecation warning gets raised. I am not sure what the semantics of TORCH_WARN_ONCE is so I am not sure how plausible this suggestion is

Reply:
Curerntly we have no way to catch this since TORCH_WARN_ONCE is not an assertion. But this one is covered in test case's log. It will be printed every time the test case is run.

Reply:
You should be able to catch python warnings using the python warnings package, as long as the function consistently raises a warning.  For example: https://github.com/pytorch/pytorch/blob/af4d6120bda2cb930b9b51b69a776ebbb9203616/test/test_torch.py#L1582-L1585

Reply:
Resolved by using assertWarnsRegex